x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/ClusterPrivilegeResolver.java

```diff
@@ -277,6 +278,11 @@ public class ClusterPrivilegeResolver {
         "manage_search_application",
         MANAGE_SEARCH_APPLICATION_PATTERN
     );
+
+    public static final NamedClusterPrivilege MANAGE_QUERY_RULES = new ActionClusterPrivilege(
```

Comment:
I'd especially like feedback from @ioanatia here - Is a "manage query rules" cluster privilege sufficient?

Reply:
If it was me, I would actually just use the `manage` cluster privilege for now because this feature is behind a feature flag and it's disabled by default. So there's no immediate need to add a cluster privilege.
The `manage` cluster privilege includes all the cluster privileges except `manage_security`, so it is perfectly fine.
We can then add this as a separate cluster privilege and have a review from the Elasticsearch security team.
The `manage` cluster privilege would most likely include this new cluster privilege.

One unknown that I have would be what would happen if at some point we decide to remove or rename this `manage_query_rules` privilege - if a cluster has an API key that uses it, when it gets upgraded to a new version where this privilege does not exist, what would the behaviour of the API key be? We had a similar conversation when we thought about removing the cluster privilege we have for behavioral analytics and rely on the new workflow restrictions.

So instead of trying to answer questions like these right away, I would just not focus on it at the moment and use the `manage` cluster privilege for now.

Reply:
Sounds reasonable, thank you. I'll update this 👍